File: java/coppa/src/test/java/com/just10/coppa/COPPATests.java
/**
 Copyright 2016 Just10 Inc.

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

 http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
package com.just10.coppa;

import org.junit.Test;

import java.util.Calendar;

import static com.just10.coppa.CalendarUtils.now;
import static com.just10.coppa.TestUtils.moveDate;
import static com.just10.coppa.TestUtils.toyyyyMMdd;
import static org.junit.Assert.*;
import static com.just10.coppa.COPPA.ageMeetsCOPPARequirements;

/**
 To work on unit tests, switch the Test Artifact in the Build Variants view.
 */
public class COPPATests {
    @Test
    public void testZeroYearOldDateTimeDoesntMeetsCOPPARequirements()
    {
        assertFalse (ageMeetsCOPPARequirements (CalendarUtils.now ()));
        Calendar bd = CalendarUtils.now ();
        bd.add (Calendar.DAY_OF_YEAR, -15);
        assertFalse (ageMeetsCOPPARequirements (bd));

        bd = CalendarUtils.now ();
        bd.add (Calendar.MONTH, -5);
        assertFalse (ageMeetsCOPPARequirements (bd));
    }

    @Test
    public void testOneYearOldDateTimeDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (moveDate(-1,0,0)));
        assertFalse (ageMeetsCOPPARequirements (moveDate (-1, 3, 0)));
    }

    @Test
    public void testFiveYearOldDateTimeDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (moveDate (-5, 0, 0)));
        assertFalse (ageMeetsCOPPARequirements (moveDate (-5, 6, 4)));
    }

    @Test
    public void testThirteenYearOldDateTimeDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (moveDate (-13, 0, 0)));
        assertFalse (ageMeetsCOPPARequirements (moveDate (-13, 7, 0)));
    }

    @Test
    public void testFourteenYearOldDayShortDateTimeDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (moveDate (-14, 0, 1)));
    }

    @Test
    public void testFourteenYearOldDateTimeMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-14, 0, 0)));
    }

    @Test
    public void testFourteenYearOldAndADayDateTimeMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-14, 0, -1)));
    }

    @Test
    public void testFifteenYearOldDateTimeMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-15, 0, 0)));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-15, 0, -20)));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-15, -4, -26)));
    }

    @Test
    public void testEighteenYearOldDateTimeMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-18, 0, 0)));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-18, 0, -7)));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-18, -8, -12)));
    }

    @Test
    public void testTwentyYearOldDateTimeMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-20, 0, 0)));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-20, 0, -22)));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-20, -2, -15)));
    }

    @Test
    public void testFiftyYearOldDateTimeMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-50, 0, 0)));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-50, 0, -9)));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-50, -10, 1)));
    }

    @Test
    public void testHundredYearOldDateTimeMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-100, 0, 0)));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-100, 0, -19)));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-100, -3, -21)));
    }

    // ageMeetsCOPPARequirements (long birthdate)
    @Test
    public void testZeroYearOldLongDateDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (moveDate (0, 0, 0).getTimeInMillis ()));
        assertFalse (ageMeetsCOPPARequirements (moveDate (0, 0, -15).getTimeInMillis ()));
        assertFalse (ageMeetsCOPPARequirements (moveDate (0, -5, 0).getTimeInMillis ()));
    }

    @Test
    public void testOneYearOldLongDateDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (moveDate (-1, 0, 0).getTimeInMillis ()));
        assertFalse (ageMeetsCOPPARequirements (moveDate (-1, 3, 0).getTimeInMillis ()));
    }

    @Test
    public void testFiveYearOldLongDateDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (moveDate (-5, 0, 0).getTimeInMillis ()));
        assertFalse (ageMeetsCOPPARequirements (moveDate (-5, 6, 4).getTimeInMillis ()));
    }

    @Test
    public void testThirteenYearOldLongDateDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (moveDate (-13, 0, 0).getTimeInMillis ()));
        assertFalse (ageMeetsCOPPARequirements (moveDate (-13, 7, 0).getTimeInMillis ()));
    }

    @Test
    public void testFourteenYearOldDayShortLongDateDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (moveDate (-14, 1, 0).getTimeInMillis ()));
    }

    @Test
    public void testFourteenYearOldLongDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-14, 0, 0).getTimeInMillis ()));
    }

    @Test
    public void testFourteenYearOldAndADayLongDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-14, -1, 0).getTimeInMillis ()));
    }

    @Test
    public void testFifteenYearOldLongDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-15, 0, 0).getTimeInMillis ()));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-15, 0, -20).getTimeInMillis ()));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-15, -4, -12).getTimeInMillis ()));
    }

    @Test
    public void testEighteenYearOldLongDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-18, 0, 0).getTimeInMillis ()));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-18, 0, -7).getTimeInMillis ()));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-18, -8, -12).getTimeInMillis ()));
    }

    @Test
    public void testTwentyYearOldLongDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-20, 0, 0).getTimeInMillis ()));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-20, 0, -25).getTimeInMillis ()));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-20, -2, -15).getTimeInMillis ()));
    }

    @Test
    public void testFiftyYearOldLongDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-50, 0, 0).getTimeInMillis ()));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-50, 0, -9).getTimeInMillis ()));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-50, -10, 1).getTimeInMillis ()));
    }

    @Test
    public void testHundredYearOldLongDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (moveDate (-100, 0, 0).getTimeInMillis ()));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-100, 0, -19).getTimeInMillis ()));
        assertTrue (ageMeetsCOPPARequirements (moveDate (-100, -3, -21).getTimeInMillis ()));
    }

    // ageMeetsCOPPARequirements (string yyyymmdd)
    @Test
    public void testZeroYearOldStringDateDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (toyyyyMMdd (now ())));
        assertFalse (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (0, 0, -15))));
        assertFalse (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (0, -5, 0))));
    }

    @Test
    public void testOneYearOldStringDateDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-1, 0, 0))));
        assertFalse (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-1, 3, 0))));
    }

    @Test
    public void testFiveYearOldStringDateDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-5, 0, 0))));
        assertFalse (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-5, 6, 4))));
    }

    @Test
    public void testThirteenYearOldStringDateDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-13, 0, 0))));
        assertFalse (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-13, 7, 0))));
    }

    @Test
    public void testFourteenYearOldDayShortStringDateDoesntMeetsCOPPARequirements ()
    {
        assertFalse (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-14, 0, 1))));
    }

    @Test
    public void testFourteenYearOldStringDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-14, 0, 0))));
    }

    @Test
    public void testFourteenYearOldAndADayStringDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-14, 0, -1))));
    }

    @Test
    public void testFifteenYearOldStringDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-15, 0, 0))));
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-15, 0, -20))));
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-15, -4, -27))));
    }

    @Test
    public void testEighteenYearOldStringDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-18, 0, 0))));
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-18, 0, -7))));
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-18, -8, -17))));
    }

    @Test
    public void testTwentyYearOldStringDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-20, 0, 0))));
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-20, 0, -23))));
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-20, -2, -19))));
    }

    @Test
    public void testFiftyYearOldStringDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-50, 0, 0))));
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-50, 0, -13))));
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-50, -10, 1))));
    }

    @Test
    public void testHundredYearOldStringDateMeetsCOPPARequirements ()
    {
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-100, 0, 0))));
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-100, 0, -19))));
        assertTrue (ageMeetsCOPPARequirements (toyyyyMMdd (moveDate (-100, -3, 7))));
    }
}

File: bb10/Tests/src/COPPATests.cpp
/*
 * COPPATests.cpp
 *
 *  Created on: Jun 1, 2016
 *      Author: mahramf
 */

#include <QTest.h>
#include "COPPATests.h"
#include <coppa.h>
#include <time_utils.h>
#include "TestUtils.h"

using namespace just10::coppa;
using namespace just10::coppa::util;

const char * COPPATests::Name()
{
    return "COPPA Tests";
}

// AgeMeetsCOPPARequirements (QDate birthdate)
void COPPATests::testZeroYearOldDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(today()));
}

void COPPATests::testOneYearOldDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-1)));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-1, 3)));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-1, -2, -12)));
}
void COPPATests::testFiveYearOldDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-5)));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-5, 6, -4)));
}
void COPPATests::testThirteenYearOldDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-13)));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-13, 7)));
}
void COPPATests::testFourteenYearOldDayShortDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-14, 0, 1)));
}
void COPPATests::testFourteenYearOldDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-14)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-14, -3, 1)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-15, 0, 1)));
}
void COPPATests::testFourteenYearOldAndADayDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-14, 0, -1)));
}
void COPPATests::testFifteenYearOldDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-15)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-15, 0, -17)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-15, -5, -27)));
}
void COPPATests::testEighteenYearOldDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-18)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-18, 0, -7)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-18, -8, -13)));
}
void COPPATests::testTwentyYearOldDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-20)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-20, 0, -15)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-20, -3, -20)));
}
void COPPATests::testFiftyYearOldDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-50)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-50, 0, -7)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-50, -10, 3)));
}
void COPPATests::testHundredYearOldDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-100)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-100, 0, -19)));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-100, -4, -14)));
}
// AgeMeetsCOPPARequirements (long birthdate)
void COPPATests::testZeroYearOldLongDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime().toMSecsSinceEpoch()));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime(0, -5).toMSecsSinceEpoch()));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime(0, 0, -23).toMSecsSinceEpoch()));
}
void COPPATests::testOneYearOldLongDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime(-1).toMSecsSinceEpoch()));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime(-1, 3).toMSecsSinceEpoch()));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime(-1, -2, -12).toMSecsSinceEpoch()));
}
void COPPATests::testFiveYearOldLongDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime(-5).toMSecsSinceEpoch()));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime(-5, 6, -4).toMSecsSinceEpoch()));
}
void COPPATests::testThirteenYearOldLongDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime(-13).toMSecsSinceEpoch()));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime(-13, 7).toMSecsSinceEpoch()));
}
void COPPATests::testFourteenYearOldDayShortLongDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveTime(-14, 0, 1).toMSecsSinceEpoch()));
}
void COPPATests::testFourteenYearOldLongDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-14).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-14, -3, 1).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-15, 0, 1).toMSecsSinceEpoch()));
}
void COPPATests::testFourteenYearOldAndADayLongDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-14, 0, -1).toMSecsSinceEpoch()));
}
void COPPATests::testFifteenYearOldLongDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-15).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-15, 0, -17).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-15, -5, -27).toMSecsSinceEpoch()));
}
void COPPATests::testEighteenYearOldLongDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-18).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-18, 0, -7).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-18, -8, -13).toMSecsSinceEpoch()));
}
void COPPATests::testTwentyYearOldLongDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-20).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-20, 0, -15).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-20, -3, -20).toMSecsSinceEpoch()));
}
void COPPATests::testFiftyYearOldLongDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-50).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-50, 0, -7).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-50, -10, 3).toMSecsSinceEpoch()));
}
void COPPATests::testHundredYearOldLongDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-100).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-100, 0, -19).toMSecsSinceEpoch()));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveTime(-100, -4, -14).toMSecsSinceEpoch()));
}
// AgeMeetsCOPPARequirements (QString birthdate)
void COPPATests::testZeroYearOldStringDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(today().toString("yyyyMMdd")));
}
void COPPATests::testOneYearOldStringDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-1).toString("yyyyMMdd")));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-1, 3).toString("yyyyMMdd")));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-1, -2, -12).toString("yyyyMMdd")));
}
void COPPATests::testFiveYearOldStringDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-5).toString("yyyyMMdd")));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-5, 6, -4).toString("yyyyMMdd")));
}
void COPPATests::testThirteenYearOldStringDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-13).toString("yyyyMMdd")));
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-13, 7).toString("yyyyMMdd")));
}
void COPPATests::testFourteenYearOldDayShortStringDateDoesntMeetsCOPPARequirements()
{
    ASSERT_FALSE(ageMeetsCOPPARequirements(moveDate(-14, 0, 1).toString("yyyyMMdd")));
}
void COPPATests::testFourteenYearOldStringDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-14).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-14, -3, 1).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-15, 0, 1).toString("yyyyMMdd")));
}
void COPPATests::testFourteenYearOldAndADayStringDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-14, 0, -1).toString("yyyyMMdd")));
}
void COPPATests::testFifteenYearOldStringDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-15).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-15, 0, -17).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-15, -5, -27).toString("yyyyMMdd")));
}
void COPPATests::testEighteenYearOldStringDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-18).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-18, 0, -7).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-18, -8, -13).toString("yyyyMMdd")));
}
void COPPATests::testTwentyYearOldStringDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-20).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-20, 0, -15).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-20, -3, -20).toString("yyyyMMdd")));
}
void COPPATests::testFiftyYearOldStringDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-50).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-50, 0, -7).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-50, -10, 3).toString("yyyyMMdd")));
}
void COPPATests::testHundredYearOldStringDateMeetsCOPPARequirements()
{
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-100).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-100, 0, -19).toString("yyyyMMdd")));
    ASSERT_TRUE(ageMeetsCOPPARequirements(moveDate(-100, -4, -14).toString("yyyyMMdd")));
}

